src/blockchain/transactions.rs

```diff
@@ -58,111 +35,107 @@ impl TxSignature {
     }
 }
 
-impl Transaction for TxSignature {
-    fn execute(&self, context: TransactionContext) -> ExecutionResult {
-        // TODO Checks that transaction author is validator.
-        let tx = &self.transaction;
-        let schema = BtcAnchoringSchema::new(context.fork());
-        // Checks that the number of signatures is sufficient to spend.
+/// Exonum BTC anchoring transactions.
+#[exonum_service]
+pub trait Transactions {
+    /// Sign a single input of the anchoring transaction proposal.
+    fn sign_input(&self, context: TransactionContext, arg: SignInput)
+        -> Result<(), ExecutionError>;
+}
+
+impl Transactions for BtcAnchoringService {
+    fn sign_input(
```

Comment:
Nit: there are so many exit conditions in this function, maybe add comments what exactly we are checking for every one of them?